### ISSUE TITLE
build Stracciatella with Visual 2010 Express

### ISSUE THREAD

**Issue by JAsmine-ja2** [ja2-stracciatella/ja2-stracciatella#301]
I've been trying to build Stracciatella with Visual C++ 2010 Express.
Building always fails with this log:

1>OppList.obj : error LNK2001: Nicht aufgelöstes externes Symbol ""void __cdecl MPrintStat(int,int,wchar_t const * const,int)" (?MPrintStat@@YAXHHQB_WH@Z)".

1>OppList.obj : error LNK2001: Nicht aufgelöstes externes Symbol ""void __cdecl MPrintStat(int,int,wchar_t const * const,wchar_t const * const)" (?MPrintStat@@YAXHHQB_W0@Z)".

1>OppList.obj : error LNK2001: Nicht aufgelöstes externes Symbol ""void __cdecl MPrintStat(int,int,wchar_t const * const,void const * const)" (?MPrintStat@@YAXHHQB_WQBX@Z)".

1>OppList.obj : error LNK2001: Nicht aufgelöstes externes Symbol ""void __cdecl MPrintStat(int,int,wchar_t const * const,int,int)" (?MPrintStat@@YAXHHQB_WHH@Z)".

1>OppList.obj : error LNK2001: Nicht aufgelöstes externes Symbol ""void __cdecl MHeader(int,int,wchar_t const * const)" (?MHeader@@YAXHHQB_W@Z)".

1>C:\spiele\Modding\Source\ja2-stracciatella-master\_build\solution-vs10e\\..\..\ja2-vs10e-Release.exe : fatal error LNK1120: 5 nicht aufgelöste externe Verweise.
========== Erstellen: 0 erfolgreich, Fehler bei 1, 0 aktuell, 0 übersprungen ==========

Now, I'm quite sure that the problem is only on my side and I'm doing something wrong due to lack of experience. I also don't want to turn this into support forum for MS Visual Studio.
But, if anyone has an idea what could be wrong, I would appreciate it.

**Comment by selaux** [comment]
`MPrintStat` is part of the new Debug Pages. They might not be included in the `sln` included in the repo. Can you try adding `Build/Utils/Debug_Pages.h` and `Build/Utils/Debug_Pages.cc` manually to your project and rebuild?

**Comment by JAsmine-ja2** [comment]
That works!!! :+1: 
Thanks a lot!

**Comment by selaux** [comment]
Can you check if the branch from https://github.com/ja2-stracciatella/ja2-stracciatella/pull/306 works out of the box?

**Comment by JAsmine-ja2** [comment]
I checked and it works - the exe is built. Here is the log:
[vs2010.txt](https://github.com/ja2-stracciatella/ja2-stracciatella/files/256653/vs2010.txt)

- There are some warnings, but those were already there before  this branch.
- The exe only builds if anti-virus software is disabled, but that seems to be a specific issue of visual 2010 and that problem was present before. Nothing can be done about this.
- The exe starts. New game can be started. Intro movies play. Laptop and strategic map can be viewed. As soon as you try to read the IMP mail, the game crashes. But that happens without this branch as well. Savegames from later in the game can be loaded. So, the mail somehow crashes the game.

**Comment by derSteFfi** [comment]
can you obtain a backtrace for that in any way?

**Comment by JAsmine-ja2** [comment]
I'm sorry, but what is a backtrace and how do I obtain it?

**Comment by selaux** [comment]
Thats nothing to be sorry about :wink:. When running the application from Visual Studio you will be able to interrupt the program when the crash occurs. In the bottom you can then see the backtrace (Aufrufliste in your case). I could reproduce this. Your backtrace should read:

```
>	ja2-vs10e-Debug.exe!PreProcessEmail(Email * m) Zeile 2412	C++
 	ja2-vs10e-Debug.exe!DisplayEmailMessage(Email * m) Zeile 997	C++
 	ja2-vs10e-Debug.exe!HandleEmail() Zeile 421	C++
 	ja2-vs10e-Debug.exe!HandleLapTopHandles() Zeile 985	C++
 	ja2-vs10e-Debug.exe!LaptopScreenHandle() Zeile 1166	C++
 	ja2-vs10e-Debug.exe!GameLoop() Zeile 299	C++
 	ja2-vs10e-Debug.exe!MainLoop(int msPerGameCycle) Zeile 248	C++
 	ja2-vs10e-Debug.exe!SDL_main(int argc, char * * argv) Zeile 511	C++

```

**Comment by wendles** [comment]
Just jumping in here to say the crash (at least with VS2015C) is a runtime check about stack corruption around the 'str' variable, in PreProcessEmail. There's a little loop where the code generates the e-mail messages. I have not figured out a fix yet, for now I just commented out that loop so the game would create the IMP link in the web browser. Naturally doing so means all e-mail bodies are blank.

**Comment by derSteFfi** [comment]
Is #306 good enough to merge it then anyway?

**Comment by JAsmine-ja2** [comment]
I can still confirm that #306 allows VS2010 to build Stracciatella without crashing. I have built several times.
The result isn't flawless, but better then nothing (before #306 )

**Comment by selaux** [comment]
It builds, but isn't playable. That is why this issue is still open, or should we create a new one for that?

**Comment by vancv** [comment]
I fixed stack corruption issue with #311

**Comment by derSteFfi** [comment]
jasmine, can you too confirm #311? Than we can merge these two.
